src/App.tsx

```diff
@@ -1,24 +1,272 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import { TodoItem } from './components/TodoItem';
+import { Status } from './types/Status';
+import { Filter } from './components/TodoFilter';
 
-const USER_ID = 0;
+const USER_ID = 11498;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeFilter, setActiveFilter] = useState(Status.All);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [inputValue, setInputValue] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [wasThereAdding, setWasThereAdding] = useState(false);
+  const [areAllCompletedDeleting, setAreAllCompletedDeleting] = useState(false);
+  const [areAllToggling, setAreAllToggling] = useState(false);
+
+  const completedTodos: Todo[] = [];
+  const activeTodos: Todo[] = [];
+  let visibleTodos: Todo[] = [];
+
+  todos.forEach(todo => {
+    if (todo.completed) {
+      completedTodos.push({ ...todo });
+    } else {
+      activeTodos.push({ ...todo });
+    }
+  });
+
+  switch (activeFilter) {
+    case Status.Active:
+      visibleTodos = activeTodos;
+      break;
+
+    case Status.Completed:
+      visibleTodos = completedTodos;
+      break;
+
+    case Status.All:
+    default:
+      visibleTodos = todos;
+  }
+
+  const areThereCompleted = completedTodos.length !== 0;
```

Comment:
it's not clear what does `areThereCompleted` stay for, pls give it more clear name like `completedExist` or whatever